good.py

```
@@ -0,0 +1,140 @@
+memory: Optional[ChatMemoryBuffer] = None
+
+
+def remove_folder(folder_path: str) -> None:
+    """Remove a folder and its contents if it exists."""
+    try:
+        if os.path.exists(folder_path):
+            shutil.rmtree(folder_path)
+            logger.info("Removed folder: %s", folder_path)
+        else:
+            logger.info("Folder does not exist: %s", folder_path)
+    except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)